Access control: Add metrics

donomii opened the pull request:
<!--

Thank you for sending a pull request! Here are some tips:

1. If this is your first time, please read our contribution guide at https://github.com/grafana/grafana/blob/main/CONTRIBUTING.md

2. Ensure you include and run the appropriate tests as part of your Pull Request.

3. In a new feature or configuration option, an update to the documentation is necessary. Everything related to the documentation is under the docs folder in the root of the repository.

4. If the Pull Request is a work in progress, make use of GitHub's "Draft PR" feature and mark it as such.

5. If you can not merge your Pull Request due to a merge conflict, Rebase it. This gets it in sync with the main branch.

6. Name your PR as "<FeatureArea>: Describe your change", e.g. Alerting: Prevent race condition. If it's a fix or feature relevant for the changelog describe the user impact in the title. The PR title is used to auto-generate the changelog for issues marked with the "add to changelog" label.

-->

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

<!--

- Automatically closes linked issue when the Pull Request is merged.

Usage: "Fixes #<issue number>", or "Fixes (paste link of issue)"

-->

Fixes https://github.com/grafana/grafana-enterprise/issues/1247

**Special notes for your reviewer**:

Branched from main and cherry-picked patches

Commit:
Add metrics for enterprise access package, instrument evaluator

This patch adds metrics to support instrumenting the accesscontrols package.

It also isntruments the accesscontrol evaluator and the ossaccesscontrol permissions function.

Commit:
Add metrics for enterprise access package, instrument evaluator

This patch adds metrics to support instrumenting the accesscontrols package.

It also isntruments the accesscontrol evaluator and the ossaccesscontrol permissions function.

Commit:
Update pkg/infra/metrics/metrics.go

Co-authored-by: Vardan Torosyan <vardants@gmail.com>

Commit:
Update pkg/infra/metrics/metrics.go

Co-authored-by: Vardan Torosyan <vardants@gmail.com>

Commit:
Move enterprise only metrics to enterprise repo

Commit:
Move enterprise only metrics to enterprise repo

thisisobate committed:
Slider: updates story from knobs to control (#33983)

sakjur commented on pkg/services/accesscontrol/evaluator/evaluator.go:
```diff
@@ -5,13 +5,18 @@ import (
 // Evaluate evaluates access to the given resource, using provided AccessControl instance.
 // Scopes are evaluated with an `OR` relationship.
 func Evaluate(ctx context.Context, ac accesscontrol.AccessControl, user *models.SignedInUser, action string, scope ...string) (bool, error) {
+	timer := prometheus.NewTimer(metrics.MAccessPermissionsSummary)
```
Won't this be registered twice per operation if we do it both in Evaluate and GetUserPermissions?

donomii replied:
I think this use is ok.  NewTimer doesn't register a metric, it does the measurement and writes it to metrics.MAccessPermissionsSummary, which is registered elsewhere.  timer is local to this call and is destroyed after every call, MAccessPermissionsSummary is the persistent metric. 

I hope.

sakjur reviewed (commented):
If you have time for it, in general I'd suggest having the counters live inside of the `pkg/services/accesscontrol` domain somewhere (say `pkg/services/accesscontrol/metrics` or even in the root) since we're moving more into package by service rather than package by layer.

That said, this is also good. I asked a question about possible double reporting that I'd like an answer to before approving, but that's the only thing 🙂

ivanahuckova committed:
Logs: Store queries that were run to get log results (#33971)

* Store queries that were run to get log results

* Fix typing

* Update public/app/features/explore/Logs.tsx

Co-authored-by: Gábor Farkas <gabor.farkas@gmail.com>

* Add changes based on feedback

Co-authored-by: Gábor Farkas <gabor.farkas@gmail.com>

kylebrandt committed:
Alerting/Chore: Backend remove def_ columns from instance (#33875)

rename def_uid and def_org_id to rule_uid and rule_org_id on the alert_instance table and drops the definition table.

hugohaggmark committed:
Urls: Parsing key values will not convert to numbers (#33982)

* Urls: Parsing key values will not convert to numbers

* Tests: fixes broken tests

* Tests: fixes broken test

* Tests: fixes broken test

* Chore: forcing drone to rebuild

owen-d committed:
Alerting/allow empty receiver (#33962)

* simplifies yaml unmarshaling: PostableApiReceiver

* allow empty receiver type

* allows name only receivers (blackhole)

* better receiver type parsing

* linting

ivanahuckova committed:
Fix running of duplicite queries when clicking on Older logs button (#33992)

Clarity-89 committed:
Grafana-UI: Add className to TagsInput (#33944)

* Grafana-UI: Add className to TagsInput

* Do not add existing tags

idafurjes committed:
32540: Add org users with pagination (#33788)

* Add model for search org user and add handler for dispatch

* 32540_org_users_with_pagination: Add endpoint for search org users

* 32540_org_users_with_pagination: Add test for org user search handler

* 32540_org_users_with_pagination: fix indentation

* 32540_org_users_with_pagination: Remove newline

* 32540_org_users_with_pagination: Remove empty line

* 32540_org_users_with_pagination: Fix indentation

* Update pkg/api/org_users.go

Co-authored-by: Arve Knudsen <arve.knudsen@gmail.com>

* Update pkg/api/org_users.go

Co-authored-by: Arve Knudsen <arve.knudsen@gmail.com>

* Update pkg/models/org_user.go

Co-authored-by: Arve Knudsen <arve.knudsen@gmail.com>

* Update pkg/api/org_users_test.go

Co-authored-by: Arve Knudsen <arve.knudsen@gmail.com>

* Update pkg/api/org_users_test.go

Co-authored-by: Arve Knudsen <arve.knudsen@gmail.com>

* 32540_org_users_with_pagination: Use hs.SQLStore.SearchOrgUsers instead of bus

* Add model for search org user and add handler for dispatch

* 32540_org_users_with_pagination: Add endpoint for search org users

* 32540_org_users_with_pagination: Add test for org user search handler

* 32540_org_users_with_pagination: fix indentation

* 32540_org_users_with_pagination: Remove newline

* 32540_org_users_with_pagination: Remove empty line

* 32540_org_users_with_pagination: Fix indentation

* Update pkg/api/org_users.go

Co-authored-by: Arve Knudsen <arve.knudsen@gmail.com>

* Update pkg/api/org_users.go

Co-authored-by: Arve Knudsen <arve.knudsen@gmail.com>

* Update pkg/models/org_user.go

Co-authored-by: Arve Knudsen <arve.knudsen@gmail.com>

* Update pkg/api/org_users_test.go

Co-authored-by: Arve Knudsen <arve.knudsen@gmail.com>

* Update pkg/api/org_users_test.go

Co-authored-by: Arve Knudsen <arve.knudsen@gmail.com>

* 32540_org_users_with_pagination: Use hs.SQLStore.SearchOrgUsers instead of bus

* 32540_org_users_with_pagination: Add test for the sqlstore

* 32540_org_users_with_pagination: Fix sqlstore test

* Update pkg/api/org_users.go

Co-authored-by: Arve Knudsen <arve.knudsen@gmail.com>

* Update pkg/api/org_users_test.go

Co-authored-by: Arve Knudsen <arve.knudsen@gmail.com>

* Update pkg/api/org_users_test.go

Co-authored-by: Arve Knudsen <arve.knudsen@gmail.com>

* Update pkg/services/sqlstore/org_users.go

Co-authored-by: Arve Knudsen <arve.knudsen@gmail.com>

* Update pkg/services/sqlstore/org_users.go

Co-authored-by: Arve Knudsen <arve.knudsen@gmail.com>

* Update pkg/services/sqlstore/org_test.go

Co-authored-by: Arve Knudsen <arve.knudsen@gmail.com>

* Update pkg/services/sqlstore/org_test.go

Co-authored-by: Arve Knudsen <arve.knudsen@gmail.com>

* 32540: Fix search org users method

* 32540: Fix sqlstore test

* 32540: Fix go-lint

Co-authored-by: Arve Knudsen <arve.knudsen@gmail.com>

dprokop committed:
Do not crash timeseries panel if there is no time series data in the response (#33993)

hugohaggmark committed:
OrgPicker: Fixes so organisations load (#33990)

* OrgPicker: Fixes so organisations load

* Update public/app/core/components/Select/OrgPicker.tsx

Co-authored-by: Alex Khomenko <Clarity-89@users.noreply.github.com>

* Chore: updates after PR comments

Co-authored-by: Alex Khomenko <Clarity-89@users.noreply.github.com>

Commit:
Add separate counter for evaluation time

Commit:
Add separate counter for evaluation time

donomii commented:
> If you have time for it, in general I'd suggest having the counters live inside of the `pkg/services/accesscontrol` domain somewhere (say `pkg/services/accesscontrol/metrics` or even in the root) since we're moving more into package by service rather than package by layer.

Happy to move it, I'll make another issue for it, I'm out of time today

julienduchesne committed:
Enterprise changes to the Drone pipelines (#33773)

* Enterprise changes to the Drone pipelines
This is basically a no-op in this repository, except for the fact that the grafanabot personal access token will now be fetched from Vault instead of repository secrets
This will pave the way for us to fetch all secrets from Vault

* Update star files from enterprise

* Add missingn newline

thisisobate committed:
Spinner: updates story from knobs to control (#33988)

owen-d committed:
bumps am & common libs with json tag support (#33995)

aocenas committed:
NodeGraph: Exploration mode (#33623)

* Add exploration option to node layout

* Add hidden node count

* Add grid layout option

* Fix panning bounds calculation

* Add legend with sorting

* Allow sorting on any stats or arc value

* Fix merge

* Make sorting better

* Reset focused node on layout change

* Refactor limit hook a bit

* Disable selected layout button

* Don't show markers if only 1 node is hidden

* Move legend to the bottom

* Fix text backgrounds

* Add show in graph layout action in grid layout

* Center view on the focused node, fix perf issue when expanding big graph

* Limit the node counting

* Comment and linting fixes

* Bit of code cleanup and comments

* Add state for computing layout

* Prevent computing map with partial data

* Add rollup plugin for worker

* Add rollup plugin for worker

* Enhance data from worker

* Fix perf issues with reduce and object creation

* Improve comment

* Fix tests

* Css fixes

* Remove worker plugin

* Add comments

* Fix test

* Add test for exploration

* Add test switching to grid layout

* Apply suggestions from code review

Co-authored-by: Zoltán Bedi <zoltan.bedi@gmail.com>

* Remove unused plugin

* Fix function name

* Remove unused rollup plugin

* Review fixes

* Fix context menu shown on layout change

* Make buttons bigger

* Moved NodeGraph to core grafana

Co-authored-by: Zoltán Bedi <zoltan.bedi@gmail.com>

thisisobate committed:
RangeSlider: updates story from knobs to control (#33980)

* RangeSlider: updates story from knobs to control

* changed the orientation control to type Select

* removes tooltipAlwaysVisible control since it crashes the story

ifrost committed:
Explore: Match queries that are longer than defined Graphite/Loki mapping (#34003)

kostrse committed:
AzureMonitor: Azure settings in Grafana server config (#33728)

* Azure cloud settings

* Fix typos

* Grouped Azure settings

* Doc fixes

* Some settings are not needed

* Updated cloud name aliases

AgnesToulet committed:
Rendering: add CSV support (#33729)

* Rendering: add CSV rendering support

* Rendering: save csv files into a separate folder

* add missing field

* Renderer: get filename from renderer plugin

* apply PR suggestions

* Rendering: remove old PhantomJS error

* Rendering: separate RenderCSV and Render functions

* fix alerting test

* Rendering: fix handling error in HTTP mode

* apply PR feedback

* Update pkg/services/rendering/http_mode.go

Co-authored-by: Joan López de la Franca Beltran <joanjan14@gmail.com>

* apply PR feedback

* Update rendering metrics with type label

* Rendering: return error if not able to parse header

* Rendering: update grpc generated file

* Rendering: use context.WithTimeout to render CSV too

Co-authored-by: Joan López de la Franca Beltran <joanjan14@gmail.com>

ifrost committed:
Docs: Add information about Graphite/Loki integration (#33916)

* Add information about Graphite/Loki integration

* Inline text

* Update docs/sources/datasources/graphite.md

Co-authored-by: achatterjee-grafana <70489351+achatterjee-grafana@users.noreply.github.com>

Co-authored-by: achatterjee-grafana <70489351+achatterjee-grafana@users.noreply.github.com>

Clarity-89 committed:
Grafana-UI: Add disabled state to picker components (#34005)

* TimeRangeInput: add disabled prop

* TimeOfDayPicker: add disabled prop

* TimeZonePicker: add disabled prop

* TagsInput: add disabled prop

FZambia committed:
live: disable gzip for ws endpoints (#34015)

ryantxu committed:
Colors: default colors for strings and boolean (#33965)

FZambia committed:
live: remove demultiplexor (#34012)

mckn committed:
Alerting: adding a time picker for selecting relative time. (#33689)

* adding placeholder for relative time range.

* fixed story.

* added basic structure to handle open/close of time range picker.

* removed section from TimeOptions since it isn't used any where.

* adding mapper and tests

* move relativetimepicker to its own dir

* added some simple tests.

* changed test.

* use relativetimerangeinput

* redo state management

* refactored the tests.

* replace timerange with relativetimerange

* wip

* wip

* did some refactoring.

* refactored time option formatting.

* added proper formatting and display of time range.

* add relative time description, slight refactor of height

* fixed incorrect import.

* added validator and changed formatting.

* removed unused dep.

* reverted back to internal function.

* fixed display of relative time range picker.

* fixed failing tests.

* fixed parsing issue.

* fixed position of time range picker.

* some more refactorings.

* fixed validation of really big values.

* added another test.

Co-authored-by: Peter Holmberg <peter.hlmbrg@gmail.com>

marefr committed:
Chore: Upgrade grafana-plugin-sdk-go to v0.96.0 (#34010)

Commit:
Merge branch 'jprice/Access-control-add-metrics_main' of https://github.com/donomii/grafana into jprice/Access-control-add-metrics_main
